src/functionalTest/java/uk/gov/hmcts/reform/waworkflowapi/controllers/IdempotencyCheckTest.java

```diff
@@ -0,0 +1,221 @@
+package uk.gov.hmcts.reform.waworkflowapi.controllers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.assertj.core.api.Assertions;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.waworkflowapi.SpringBootFunctionalBaseTest;
+import uk.gov.hmcts.reform.waworkflowapi.clients.model.DmnValue;
+import uk.gov.hmcts.reform.waworkflowapi.clients.model.SendMessageRequest;
+import uk.gov.hmcts.reform.waworkflowapi.clients.model.idempotentkey.IdempotentId;
+import uk.gov.hmcts.reform.waworkflowapi.clients.model.idempotentkey.IdempotentKeys;
+import uk.gov.hmcts.reform.waworkflowapi.clients.service.IdempotentKeysRepository;
+import uk.gov.hmcts.reform.waworkflowapi.utils.AuthorizationHeadersProvider;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static net.serenitybdd.rest.SerenityRest.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@Slf4j
+public class IdempotencyCheckTest extends SpringBootFunctionalBaseTest {
+
+    @Autowired
+    private AuthorizationHeadersProvider authorizationHeadersProvider;
+
+    @Autowired
+    private IdempotentKeysRepository idempotentKeysRepository;
+
+    private String serviceAuthorizationToken;
+    private String caseId;
+
+    @Before
+    public void setUp() {
+        caseId = UUID.randomUUID().toString();
+
+        serviceAuthorizationToken =
+            authorizationHeadersProvider
+                .getAuthorizationHeaders()
+                .getValue(SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    public void transition_creates_a_task_and_goes_through_external_task() {
```

Comment:
ok, I can put that FT, np.

Reply:
@raghera , yes, sure. Do you want a FT for that? It can be covered well with junit and integration tests too. 
In any case, shall we do this as part of the ticket?

Reply:
I think a functional test.  Part of this one ideally.